### PR TITLE
Deprecate default precision and scale for decimal columns

### PR DESCRIPTION
**Pull request by morozov**
Similar to the string columns (https://github.com/doctrine/dbal/issues/3263), having the DBAL define the precision and scale of decimal columns doesn't make much sense. The values of precision and scale should reflect the properties of the model represented by the schema.

### CODE REVIEW

File: src/Platforms/AbstractPlatform.php [hunk 1]
```diff
@@ -2912,12 +2912,33 @@ public function getColumnDeclarationSQL($name, array $column)
...
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/5637',
+                'Relying on the default decimal column precision is deprecated'
+                    . ', specify the precision explicitly.',
+            );
+
+            $precision = 10;
+        }
+
+        if (isset($column['scale'])) {
```
Comment (greg0ire) [inline src/Platforms/AbstractPlatform.php]: The code you removed used similar conditions for precision and scale, and the code you introduced uses `empty` for precision, and `isset` for scale. Why this change?

Comment (morozov) [reply src/Platforms/AbstractPlatform.php]: Because scale can be empty (specifically, `0`) which shouldn't trigger a deprecation notice. But now I realize that by just checking the scale for `isset()` we can break the code that relies on the `empty()` check and passes NULL.

This code can be improved.

**Review by greg0ire — APPROVED** [review]
